File: app/src/main/java/com/hypnoweb/hypnowebapp/DI/Components/AppComponent.java
package com.hypnoweb.hypnowebapp.DI.Components;

import com.hypnoweb.hypnowebapp.DI.Modules.AppModule;
import com.hypnoweb.hypnowebapp.SplashScreenActivity;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by stefan on 12/3/2017.
 */

@Component(modules = {AppModule.class})
@Singleton
public interface AppComponent {
    void inject(SplashScreenActivity splashScreenActivity);
}

File: app/src/main/java/com/hypnoweb/hypnowebapp/di/Components/AppComponent.kt
package com.hypnoweb.hypnowebapp.di.components

import android.app.Application
import com.hypnoweb.hypnowebapp.di.HypnoWebApplication
import com.hypnoweb.hypnowebapp.di.modules.*
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import dagger.android.support.DaggerApplication
import javax.inject.Singleton


@Component(modules = arrayOf(
        AndroidInjectionModule::class,
        AppModule::class,
        ViewModelModule::class,
        NetworkModule::class,
        DatabaseModule::class,
        ActivityModule::class,
        FragmentBuilderModule::class,
        NavigatorModule::class))

@Singleton
interface AppComponent : AndroidInjector<DaggerApplication> {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder
        fun build(): AppComponent
    }

    override fun inject(application: DaggerApplication)
    fun inject(hypnoWebApplication: HypnoWebApplication)

}